access/resource_sql_permissions.go

```diff
@@ -281,22 +281,15 @@ func (ta *SqlPermissions) getOrCreateCluster(clustersAPI clusters.ClustersAPI) (
 		LongTermSupport: true,
 	})
 	nodeType := clustersAPI.GetSmallestNodeType(compute.NodeTypeRequest{LocalDisk: true})
+	// The cluster used for setting table ACLs is a single node cluster with USER_ISOLATION data security mode.
 	aclCluster, err := clustersAPI.GetOrCreateRunningCluster(
 		"terraform-table-acl", clusters.Cluster{
 			ClusterName:            "terraform-table-acl",
 			SparkVersion:           sparkVersion,
 			NodeTypeID:             nodeType,
 			AutoterminationMinutes: 10,
-			DataSecurityMode:       "LEGACY_TABLE_ACL",
-			// TODO: return back after backend fix is rolled out
-			NumWorkers: 1,
-			// SparkConf: map[string]string{
-			// 	"spark.databricks.cluster.profile": "singleNode",
-			// 	"spark.master":                     "local[*]",
-			// },
-			// CustomTags: map[string]string{
-			// 	"ResourceClass": "SingleNode",
-			// },
+			DataSecurityMode:       "USER_ISOLATION",
+			IsSingleNode:           true,
```

Comment:
IsSingleNode should work only if you specify `kind` that is in preview... I think that it's better to stuck with the "traditional way" via spark conf, etc.  Also, right now there is a configuration drift (we have PR for it)